new service v1.0

a5x55555 opened the pull request.

a5x55555 committed:
new service v1.0

a5x55555 committed:
fixes

VLiamin commented on src/FamilyService.Data.Provider/IDataProvider.cs:
```diff
@@ -0,0 +1,13 @@
+﻿using LT.DigitalOffice.Kernel.Enums;
+using LT.DigitalOffice.Kernel.Attributes;
+using LT.DigitalOffice.Kernel.EFSupport.Provider;
+
```
empty line

VLiamin commented:
May be add .editorconfig here?

a5x55555 committed:
family controller

a5x55555 committed:
fix

a5x55555 committed:
editorconfig & gitignore

VLiamin commented on src/FamilyService.Data.Provider.MsSql.Ef/FamilyServiceDbContext.cs:
```diff
@@ -0,0 +1,50 @@
+
+        public object MakeEntityDetached(object obj)
+        {
+            Entry(obj).State = EntityState.Detached;
```
May be make 2 spaces instead of 4?

VLiamin replied:
In all cases

a5x55555 replied:
What's the difference?

a5x55555 replied:
> May be make 2 spaces instead of 2?

Maybe 4 instead of 2?

VLiamin commented:
May be add links on editor and gitignore to LT.DigitalOffice.FamilyService.sln?

a5x55555 committed:
Update LT.DigitalOffice.FamilyService.sln

a5x55555 committed:
fix tabs and usings

MariyaSin commented on Common.Configurations.targets:
```diff
@@ -0,0 +1,41 @@
+    <PackageReference Include="LT.DigitalOffice.Kernel.BrokerSupport" Version="1.0.6" />
+    <PackageReference Include="LT.DigitalOffice.Kernel.EndpointSupport" Version="1.0.0" />
+    <PackageReference Include="LT.DigitalOffice.Kernel.EFSupport" Version="1.0.1" />
+    <PackageReference Include="LT.DigitalOffice.Models" Version="1.1.6" />
```
update all packages to latest

MariyaSin commented on src/FamilyService/FamilyService.csproj:
```diff
@@ -0,0 +1,45 @@
+    <PackageReference Include="AspNetCore.HealthChecks.UI.Client" Version="5.0.1" />
+    <PackageReference Include="LT.DigitalOffice.Kernel" Version="1.1.63" />
+    <PackageReference Include="LT.DigitalOffice.Kernel.BrokerSupport" Version="1.0.7" />
+    <PackageReference Include="LT.DigitalOffice.Kernel.EFSupport" Version="1.0.1" />
```
you have kernel packages in target

MariyaSin commented on src/FamilyService/appsettings.Development.json:
```diff
@@ -0,0 +1,19 @@
+  "Kestrel": {
+    "EndPoints": {
+      "Http": {
+        "Url": "http://[:]:9814"
```
9862

a5x55555 committed:
fixes

- package references were fixed
- packages in Common.Configuration.targets were updated
- usings were fixed

a5x55555 committed:
db models & an initial migration

a5x55555 committed:
db models fixes

a5x55555 committed:
fix

a5x55555 committed:
db models & migration fixes

MariyaSin commented on src/FamilyService.Data/Provider/Migrations/202209091712_InitialTables.cs:
```diff
@@ -0,0 +1,31 @@
+          Id = table.Column<Guid>(nullable: false),
+          Name = table.Column<string>(nullable: false, maxLength: 45),
+          Gender = table.Column<int>(nullable: false),
+          BirthDate = table.Column<DateTime>(nullable: false),
```
DateOfBirth

a5x55555 committed:
db models & migration fixes v2

a5x55555 committed:
fix

a5x55555 committed:
fix using statements

a5x55555 committed:
fix

a5x55555 committed:
Create child endpoint

a5x55555 committed:
validation enhancement

checks if this child was already mentioned

a5x55555 committed:
fix

'Child' folder was added to Models.Dto.Requests.

VLiamin commented on Common.Configurations.targets:
```diff
@@ -0,0 +1,41 @@
+    <PackageReference Include="LT.DigitalOffice.Kernel.BrokerSupport" Version="1.0.7" />
+    <PackageReference Include="LT.DigitalOffice.Kernel.EndpointSupport" Version="1.0.0" />
+    <PackageReference Include="LT.DigitalOffice.Kernel.EFSupport" Version="1.0.1" />
+    <PackageReference Include="LT.DigitalOffice.Models" Version="1.1.11" />
```
May be change libraries on DigitalOffice.Models?

VLiamin replied:
And others libraries

VLiamin commented on src/FamilyService.Business/Commands/Child/CreateChildCommand.cs:
```diff
@@ -0,0 +1,76 @@
+      OperationResultResponse<Guid?> response = new();
+
+      response.Body = await _childRepository.CreateAsync(_dbChildMapper.Map(request,
+         _httpContextAccessor.HttpContext.GetUserId()));
```
spaces

VLiamin commented on src/FamilyService.Data/ChildRepository.cs:
```diff
@@ -0,0 +1,45 @@
+    {
+      return _provider.Children
+        .AnyAsync(c => c.ParentUserId == ParentUserId
+        && c.Name == Name
```
May be add tab?

VLiamin commented on src/FamilyService.Models.Db/DbChild.cs:
```diff
@@ -0,0 +1,40 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+
```
empty line

VLiamin commented on src/FamilyService.Validation/Child/CreateChildRequestValidator.cs:
```diff
@@ -0,0 +1,41 @@
+
+      RuleFor(r => r)
+        .MustAsync(async (r, _, _) => !await _userRepository
+        .DoesValueExist(r.ParentUserId, r.Name, r.DateOfBirth.Date))
```
May be add tab?

a5x55555 committed:
fixes

a5x55555 committed:
CreateChildCommand fix

MariyaSin commented on src/FamilyService.Models.Db/DbChild.cs:
```diff
@@ -0,0 +1,39 @@
+
+namespace LT.DigitalOffice.FamilyService.Models.Db
+{
+  public partial class DbChild
```
why partial?

MariyaSin commented on src/FamilyService.Data.Provider/IDataProvider.cs:
```diff
@@ -0,0 +1,12 @@
+  [AutoInject(InjectType.Scoped)]
+  public interface IDataProvider : IBaseDataProvider
+  {
+    //DbSet<DbMembers> Members{get; set;}
```
?

MariyaSin commented on src/FamilyService.Data.Provider.MsSql.Ef/FamilyServiceDbContext.cs:
```diff
@@ -0,0 +1,50 @@
+    {
+    }
+
+    //public DbSet<DbMembers> Members{get; set;}
```
?

MariyaSin commented on src/FamilyService.Data/Provider/IDataProvider.cs:
```diff
@@ -0,0 +1,14 @@
+  [AutoInject(InjectType.Scoped)]
+  public interface IDataProvider : IBaseDataProvider
+  {
+    DbSet<DbChild> Children { get; set;}
```
DbSet<DbChild> Children { get; set; }

MariyaSin commented on src/FamilyService.Data/Provider/Migrations/202209091712_InitialTables.cs:
```diff
@@ -0,0 +1,36 @@
+        columns: table => new
+        {
+          Id = table.Column<Guid>(nullable: false),
+          Name = table.Column<string>(nullable: false, maxLength: 45),
```
do not set maxLength

MariyaSin commented on src/FamilyService.Data/Provider/Migrations/202209091712_InitialTables.cs:
```diff
@@ -0,0 +1,36 @@
+          Name = table.Column<string>(nullable: false, maxLength: 45),
+          Gender = table.Column<int>(nullable: false),
+          DateOfBirth = table.Column<DateTime>(nullable: false),
+          Info = table.Column<string>(nullable: false, maxLength: 300),
```
do not set maxLength
nullable true

MariyaSin commented on src/FamilyService.Validation/Child/CreateChildRequestValidator.cs:
```diff
@@ -0,0 +1,41 @@
+    {
+      RuleFor(r => r.Name)
+        .Cascade(CascadeMode.Stop)
+        .MinimumLength(1).WithMessage("Minimum lenght is 1.")
```
use NotEmpty

when you use the minimum length, the validator will not catch the string with white space

MariyaSin commented on src/FamilyService.Validation/Child/CreateChildRequestValidator.cs:
```diff
@@ -0,0 +1,41 @@
+        .WithMessage("This gender is not specified.");
+
+      RuleFor(r => r.DateOfBirth)
+        .Must(d => (DateTime.UtcNow.Year - d.Year) <= 18)
```
do not include 18. 
"    < 18"

MariyaSin commented on src/FamilyService.Validation/Child/CreateChildRequestValidator.cs:
```diff
@@ -0,0 +1,41 @@
+        .WithMessage("Age mustn't be greater than 18.");
+
+      RuleFor(r => r.Info)
+        .MinimumLength(0).WithMessage("Minimum lenght is 0.")
```
the property can be empty, remove the b=validation

MariyaSin commented on src/FamilyService/Startup.cs:
```diff
@@ -0,0 +1,160 @@
+
+    #region private methods
+
+    private void ConfigureMassTransit(IServiceCollection services)
```
use from kernel

a5x55555 committed:
fixes
